docs/guides/installing-tls-certificates.md

```diff
@@ -0,0 +1,97 @@
+# Generating and installing TLS certificates
+
+- [1. Overview](#1-overview)
+- [2. Generating TLS certificates](#2-generating-tls-certifificates)
+- [3. Installing TLS certificates](#3-installing-tls-certificates)
+    - [3.1 Install at default location](#31-install-at-default-location)
+    - [3.2 Install at non-default location](#32-install-at--non--default-location)
+
+## 1. Overview
+
+This document provides information about generating and installing TLS certificates
+for running infrap4d in secure mode
+
+## 2. Generating TLS certificates
+Note: Here target name can be `dpdk` or `es2k`
+
+- Review the files `ca.conf` and `grpc-client.conf` available under
+`/usr/share/stratum/<target_name>` to verify that the configuration settings are
+as desired
+- Run the `generate-certs.sh` available under `/usr/share/stratum/<target_name>`
+
+```bash
+	cd /usr/share/stratum/<target_name>/
+
+	COMMON_NAME=<IP> ./generate-certs.sh
+	or
+	COMMON_NAME=localhost ./generate-certs.sh
+```
+
+Note: System relies on mTLS (mutual TLS) for authentication.
+
+## 3. Installing TLS certificates
+
+`infrap4d` will check for server certificates in the default location
+`/usr/share/stratum/certs/`
+
+### 3.1 Install at default location
+
+- Copy the generated `ca.crt`, `stratum.crt`, and `stratum.key` to the server
+running infrap4d:
+
+
+```bash
+mkdir -p /usr/share/stratum/certs
+cp /usr/share/stratum/<target_name>/certs/stratum.crt /usr/share/stratum/certs/
+cp /usr/share/stratum/<target_name>/certs/ca.crt /usr/share/stratum/certs/
+cp /usr/share/stratum/<target_name>/certs/stratum.key /usr/share/stratum/certs/
+```
+
+- Copy the generated `ca.crt`, `client.crt`, and `client.key` to the
+client machine
+
+
+```bash
+mkdir -p /usr/share/stratum/certs
+cp /usr/share/stratum/<target_name>/certs/ca.crt /usr/share/stratum/certs/
+cp /usr/share/stratum/<target_name>/certs/client.crt /usr/share/stratum/certs/
+cp /usr/share/stratum/<target_name>/certs/client.key /usr/share/stratum/certs/
+```
+
+### 3.2 Install at non-default location
+
+If alternate location is desired for server certificates, copy the certifactes
+at desired location and the location must be specified with following options
+when running infrap4d.
+
+```text
+-ca_cert_file=[CA certificate file]
+-server_cert_file=[Server certificate file]
+-server_key_file=[Server private key file]
+```
+
+For example:
+
+```bash
```

Comment:
The examples do not make it clear what you're doing.

Are you moving files between systems? Between directories on a shared disk?